iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/MoveFeederBay.java

```diff
@@ -0,0 +1,132 @@
+import static com.powsybl.iidm.modification.util.ModificationReports.moveFeederBayBusbarSectionReport;
+import static com.powsybl.iidm.modification.util.ModificationReports.notFoundConnectableReport;
+
+/**
+ * This modification moves a feeder bay from one busbar section to another.
+ * It identifies and updates the relevant switches and connections to achieve the move operation.
+ *
+ * @author  Ghazwa Rehili {@literal <ghazwa.rehili at rte-france.com>}
+ */
+public class MoveFeederBay extends AbstractNetworkModification {
+    private final String connectableId;
+    private final String targetBusOrBusbarSectionId;
+    private final String targetVoltageLevelId;
+    private final Terminal terminal;
+
+    /**
+     * Constructor
+     * @param connectableId non-null id of the connectable whose feeder bay will be moved (busbar sections are not accepted)
+     * @param terminal non-null id of the terminal
+     * @param targetBusbarSectionId non-null id of the target busbar section
+     * @param targetVoltageLevelId non-null id of the target voltage Level
+     */
+    public MoveFeederBay(String connectableId, String targetBusbarSectionId, String targetVoltageLevelId, Terminal terminal) {
+        this.connectableId = Objects.requireNonNull(connectableId);
+        this.targetBusOrBusbarSectionId = Objects.requireNonNull(targetBusbarSectionId);
+        this.targetVoltageLevelId = Objects.requireNonNull(targetVoltageLevelId);
+        this.terminal = Objects.requireNonNull(terminal);
+    }
+
+    @Override
+    public void apply(Network network, NamingStrategy namingStrategy, boolean throwException, ComputationManager computationManager, ReportNode reportNode) {
+        Connectable<?> connectable = network.getConnectable(connectableId);
+        if (!checkConnectable(throwException, reportNode, connectable)) {
+            return;
+        }
+        VoltageLevel voltageLevel = network.getVoltageLevel(targetVoltageLevelId);
+        if (voltageLevel.getTopologyKind() == TopologyKind.NODE_BREAKER) {
+            cleanNodeBreakerTopology(network, connectableId, reportNode);
+            createTopology(getSideFromTerminal(terminal, connectable), targetBusOrBusbarSectionId, network, voltageLevel, connectable, namingStrategy, reportNode);
+            Map<VoltageLevel, Integer> firstAvailableNodes = new HashMap<>();
+            int connectableNode = firstAvailableNodes.compute(voltageLevel, this::getNextAvailableNode);
+            terminal.getNodeBreakerView().moveConnectable(connectableNode, voltageLevel.getId());
+        } else {
+            terminal.getBusBreakerView().moveConnectable(targetBusOrBusbarSectionId, terminal.isConnected());
+        }
+    }
+
+    private int getSideFromTerminal(Terminal terminal, Connectable<?> connectable) {
+        if (connectable instanceof Injection<?>) {
+            return 0;
+        } else if (connectable instanceof Branch<?>) {
```

Comment:
If I'm not mistaken, we can also easily support 3-winding transformers.
All we have to do is to:
- add a ` || connectable instanceof ThreeWindingsTransformer` in this condition;
- handle the 3 sides in `TopologyModificationUtils#getNode`;
- (and add a unit test).

Reply:
Thanks!